### DISCUSSION TITLE
How to get the metirc of epoch?

### DISCUSSION THREAD

**Discussion by Struggle-Forever** [Lightning-AI/pytorch-lightning#13870] [General]
```
    def test_step(self, batch, batch_idx, dataloader_idx):
        f1, bert_out = self.share_val_step(batch, is_dev=False)
        metrics = {'test_f1': f1}
        self.log_dict(metrics, prog_bar=True, logger=True, on_epoch=True)

    def test_step_end(self, output_results):
        pass

    def test_epoch_end(self, test_step_outputs):
      # I wantto get epoch-level metric 

```

I want to get the epoch-level metric in **def test_epoch_end(self, test_step_outputs)**

**Comment by Struggle-Forever** [comment]
Generally，we can get  epoch-level metric via   **result = trainer.test(model, ckpt_path='best')**.   
I have a new requirement to record the epoch ， the metirc of validation set and  test set when obtaining the best model.   How do I get these metrics?
